### ISSUE TITLE
Code is moved out of default case of switch

### ISSUE THREAD

**Issue by Marcono1234** [leibnitz27/cfr#133]
# CFR version
0.150-SNAPSHOT (commit cf242d2)

# Compiler
- javac 11.0.5
- javac 14

# Description
Relates to #128

Code in the default case of a switch statement or expression with no other cases (or fall-through to default) is moved outside the switch. This is sometimes not possible, e.g. when the default case defines a local class with the same name.

Source:
```
class SwitchExpressionLocalClass {
    void test() {
        switch(0) {
            default: {
                class Test {
                    Test(int i) { }
                }
                
                new Test(1);
            }
        };
        class Test { }
        new Test();
    }
}
```

Decompiled output:
```
class SwitchExpressionLocalClass {
    SwitchExpressionLocalClass() {
    }

    /*
     * Inner class renamed, behaviour may change
     * Enabled force condition propagation
     * Lifted jumps to return sites
     */
    void test() {
        switch (0) {
            default: 
        }
        class Test_1 {
            Test_1(int n) {
            }
        }
        new Test_1(1);
        class Test {
            Test() {
            }
        }
        new Test();
    }
}
```

**Comment by Marcono1234** [comment]
The same issue also occurs when using code blocks, e.g.:
```
void test() {
    {
        class Test {
                Test(int i) { }
            }
            
            new Test(1);
    }
    
    class Test { }
    new Test();
}
```
Do you want a separate issue for this?

**Comment by leibnitz27** [comment]
TBH, this one is probably a 'never fix'.

Compare

https://github.com/leibnitz27/cfr_tests/blob/master/src_6/org/benf/cfr/tests/SwitchExpressionLocalClass.java
and
https://github.com/leibnitz27/cfr_tests/blob/master/src_6/org/benf/cfr/tests/SwitchExpressionLocalClass2.java

They generate *identical*  bytecode, as you might expect.

In the case where there's no indication that code belongs in the terminal branch, there's no reason to pull it back up (indeed most of the time it makes stuff worse).

(thus, https://github.com/leibnitz27/cfr_tests/blob/master/src_6/org/benf/cfr/tests/SwitchExpressionLocalClass3.java behaves as you're expecting).

**Comment by leibnitz27** [comment]
I'm pretty sure there's already a bug raised (almost certainly by you ;) ) which covers https://github.com/leibnitz27/cfr_tests/blob/master/src_6/org/benf/cfr/tests/SwitchExpressionLocalClass2.java , but I can't see it right now.

**Comment by leibnitz27** [comment]
Oh maybe not, given your comment above!

FWIW for the next few months progress on this is going to go right down, as I don't have a commute, and that's where I get all my lovely peace and quiet to code ;)

**Comment by Marcono1234** [comment]
Yeah, I briefly looked at the byte code and noticed that it does not provide the necessary information to easily tell where the local class is declared. So I am perfectly fine with you considering this a "never fix" (and closing this issue).

Thanks for letting me know that the process will be reduced. Apparently you have had a pretty long commute based on the amount of work you have put into this :D
Stay well :)

**Comment by leibnitz27** [comment]
As discussed - this isn't going to be fixed, apart from the cases (like #128) where it's possible to know which form must be required. (and the fix for #128 is fairly heavy handed, so I'm keen not to use it generally ;) )
